packages/eventual-send/src/index.js

```diff
@@ -304,42 +381,52 @@ export function makeHandledPromise() {
       rejectHandled,
       resolveWithPresence,
     );
+
     return handledP;
   }
 
-  HandledPromise.prototype = Promise.prototype;
-  Object.setPrototypeOf(HandledPromise, Promise);
-
   function isFrozenPromiseThen(p) {
     return (
       isFrozen(p) &&
       getPrototypeOf(p) === Promise.prototype &&
       Promise.resolve(p) === p &&
-      gopd(p, 'then') === undefined
+      getOwnPropertyDescriptor(p, 'then') === undefined
     );
   }
 
+  /** @type {import('.').HandledPromiseStaticMethods} */
```

Comment:
These used to be ambient tyes.  I think either style is alright, and I also prefer having the fact that it's an import be visible on first mention.  It's slightly less code to make the first mention be on first use (since it's only imported once), but I agree with you that if there was more than one use, I'd do the import at top.

But I'll obey this suggestion for `HandledPromiseConstructor` which *is* reused.